management/dns_update.py

```diff
@@ -785,33 +795,38 @@ def set_custom_dns_record(qname, rtype, value, action, env):
 	if made_change:
 		# serialize & save
 		write_custom_dns_config(newconfig, env)
-
 	return made_change
 
 ########################################################################
 
-def get_secondary_dns(custom_dns):
+def get_secondary_dns(custom_dns, dns_type=['_secondary_nameserver']):
+	valid_types = set(['_secondary_nameserver', '_secondary_notify_xfr'])
+	if not valid_types.issuperset(set(dns_type)):
+		raise ValueError("Valid types are one or more of the following: %s" % ", ".join(valid_types))
+
+	values = []
 	for qname, rtype, value in custom_dns:
-		if qname == "_secondary_nameserver":
-			return value
+		if qname in dns_type:
+			if isinstance(value, str):
+				values.append(value)
+	if len(values) > 0:
+		return values
 	return None
 
 def set_secondary_dns(hostname, env):
-
-	if hostname in (None, ""):
-		# Clear.
-		set_custom_dns_record("_secondary_nameserver", "A", None, "set", env)
-	else:
-		# Validate.
-		hostname = hostname.strip().lower()
+	hostnames = [item.strip().lower() for item in hostname]
+	if len(hostnames) > 0:
 		resolver = dns.resolver.get_default_resolver()
-		try:
-			response = dns.resolver.query(hostname, "A")
-		except (dns.resolver.NoNameservers, dns.resolver.NXDOMAIN, dns.resolver.NoAnswer):
-			raise ValueError("Could not resolve the IP address of %s." % hostname)
-
-		# Set.
-		set_custom_dns_record("_secondary_nameserver", "A", hostname, "set", env)
+		for item in hostnames:
+			try:
+				response = dns.resolver.query(item, "A")
+			except (dns.resolver.NoNameservers, dns.resolver.NXDOMAIN, dns.resolver.NoAnswer):
+				raise ValueError("Could not resolve the IP address of %s." % item)
+			# Set.
+		set_custom_dns_record("_secondary_nameserver", "A", {"A":hostnames}, "set", env)
```

Comment:
I don't like that either. Think it is pretty messy. Just could not figure out how to get it to work in an elegant way without a lot of collateral code changing. Probably because I am still not very familiar with the codebase.